src/services/status.ts

```diff
@@ -0,0 +1,31 @@
+import _ from 'lodash';
+import CodeforcesClient from '@acmascis/codeforces-client';
+import { ApiResponse } from '@acmascis/codeforces-client/build/interfaces/api-response.interface';
+import { Submission } from '@acmascis/codeforces-client/build/interfaces/submission.interface';
+
+export const getStatusData = async (client: CodeforcesClient, contestId: string) => {
+    const contestStatus = await client.contest.status({ contestId: contestId });
+    const handlesIds = await getHandles(contestStatus);
+    const submissions = await getSubmissions(contestStatus);
+
+    return { handlesIds, submissions };
+};
+
+const getHandles = async (contestStatus: ApiResponse<Submission[]>) => {
+    let id = 0;
+    const handlesIds: { [handle: string]: number } = {};
+    if (contestStatus.status === 'OK') {
```

Comment:
This kind of checks shouldn't be the responsibility for the dependant function, instead they should be handled once in `getStatusData` as their parent function, and then pass the `contestStatus.result` directly to them to do their processing on it.